src/main/scala/mesosphere/marathon/event/http/HttpEventStreamServlet.scala

```diff
@@ -0,0 +1,115 @@
+package mesosphere.marathon.event.http
+
+import javax.inject.{ Inject, Named }
+import javax.servlet.ServletConfig
+import javax.servlet.http.{ HttpServletResponse, HttpServlet, HttpServletRequest }
+
+import akka.actor.ActorRef
+import mesosphere.marathon.ModuleNames
+import mesosphere.marathon.event.http.HttpEventStreamActor._
+import org.eclipse.jetty.servlets.EventSource.Emitter
+import org.eclipse.jetty.servlets.{ EventSource, EventSourceServlet }
+import org.eclipse.jetty.websocket.WebSocket.Connection
+import org.eclipse.jetty.websocket.{ WebSocket, WebSocketServlet }
+
+/**
+  * Handle a WebSocket client stream by delegating events to the stream actor.
+  * @param request the initial http request.
+  * @param httpStreamActor the stream actor.
+  */
+class HttpEventWebSocketHandle(request: HttpServletRequest, httpStreamActor: ActorRef)
+    extends WebSocket.OnTextMessage with HttpEventStreamHandle {
+
+  private var connection: Option[Connection] = None
+
+  override def onMessage(data: String): Unit = {
+    httpStreamActor ! HttpEventStreamIncomingMessage(this, data)
+  }
+
+  override def onOpen(connection: Connection): Unit = {
+    this.connection = Some(connection)
+    httpStreamActor ! HttpEventStreamConnectionOpen(this)
+  }
+
+  override def onClose(closeCode: Int, message: String): Unit = {
+    httpStreamActor ! HttpEventStreamConnectionClosed(this)
+    connection = None
+  }
+
+  def sendMessage(message: String): Unit = connection.foreach { conn =>
+    if (conn.isOpen) conn.sendMessage(message)
```

Comment:
Why the check? Couldn't the connection be closed the moment after we check? `sendMessage` throws an IOException. Should we handle it?